src/frontend/screens/Settings/MapSettings/BackgroundMaps.tsx

```diff
@@ -77,43 +115,56 @@ export const BackgroundMaps: NativeNavigationComponent<"BackgroundMaps"> = ({
     }
 
     if (results.type === "success") {
-      // To do API call to import map
-      sheetRef.current?.close();
+      setModalContent("loading");
+      try {
+        await api.maps.importTileset(results.uri);
+        const list = await api.maps.getStyleList();
+        setBackgroundMapList(list);
+        setModalContent("import");
+        sheetRef.current?.close();
+      } catch (err) {
+        console.log("FAILED TO IMPORT", err);
+        setModalContent("error");
+      }
```

Comment:
there are cases when the bottom sheet doesn't close after the import finishes successfully. not sure if the issue lies here but something about the modal closing/opening logic probably needs tweaking